lambda_functions/nightly_stats_bigquery_upload/nightly_stats_bigquery_upload_lambda.py

```diff
@@ -41,20 +49,16 @@ def read_nightly_stats_from_s3(
 def delete_existing_rows_for_date(
     bigquery_client: bigquery.Client,
     table_id: str,
-    object_key: str,
+    date: str,
 ) -> None:
-    date, _extension = object_key.split('.')
-    dml_statement = f"DELETE FROM `{table_id}` WHERE date = '{date}'"
+    dml_statement = f"DELETE FROM `{table_id}` WHERE date = '{date}'"  # nosec
```

Comment:
Added `# nosec` here as it was flagged by bandit. The lambda only triggers off s3 put events, which we control in our API's scheduled tasks. This seems like a very low risk.

Reply:
bandit seems to flag any query string with variables. We should make sure to sanitize input still, but just something to be aware of.